Support .go-version file parsing

hatajoe opened the pull request:
Find executable go command from .go-version which placed on top directory
as long as when $GOCMD was not set.

hatajoe committed:
Support .go-version file parsing

Find executable go command from .go-version which placed on top directory
as long as when $GOCMD was not set.

nanasi880 commented on go/main.go:
```diff
@@ -67,6 +68,38 @@ func getenv(key string, def string) string {
+	return string(removeNewLine(out))
+}
+
+func readGoVersion(path string) string {
```
It is simpler to use this.
https://golang.org/pkg/io/ioutil/#ReadFile

nanasi880 commented on go/main.go:
```diff
@@ -67,6 +68,38 @@ func getenv(key string, def string) string {
+	return "go" + string(removeNewLine(b))
+}
+
+func removeNewLine(b []byte) []byte {
```
It is better to use this.
https://golang.org/pkg/bytes/#TrimRight

nanasi880 commented on go/main.go:
```diff
@@ -67,6 +68,38 @@ func getenv(key string, def string) string {
+}
+
+func readGoVersion(path string) string {
+	f, err := os.Open(path + "/.go-version")
```
please use this.
https://golang.org/pkg/path/filepath/#Join

nanasi880 reviewed (changes requested):
Thanks for the pull request!
I want to some request change.
Please check it.

hatajoe committed:
Fix review feedback

- use ioutil.ReadFile instead of os.Open
- use bytes.TrimRight instead of byte slicing
- use bytes.Join as filepath concatenating

ref: https://github.com/nanasi880/go-env/pull/1#pullrequestreview-231206501

hatajoe commented:
@nanasi880 Thanks for feedback. I fixed completely. Please check it out.
